### ISSUE TITLE
UsesLandlordConnection::getConnectionName() not compatible

### ISSUE THREAD

**Issue by telkins** [spatie/laravel-multitenancy#333]
I'm not 100% sure, but I believe this is an error.

The `Tenant` model declares `getConnectionName` by way of the `UsesLandlordConnection` trait and looks like so:
```php
    public function getConnectionName(): string
    {
        return $this->landlordDatabaseConnectionName();
    }
```

This differs from `Model::getConnectionName()`, which looks like so:
```php
    /**
     * Get the current connection name for the model.
     *
     * @return string|null
     */
    public function getConnectionName()
    {
        return $this->connection;
    }
```

It doesn't seem like this is a problem under "normal" conditions, but I get an error when I use another package (Caleb Porzio's Sushi) that overrides `getConnectionName()` like so:
```php
    public function getConnectionName()
    {
        return static::class;
    }
```

I get the following error when I try to use a Sushi class as a tenant:
```
Declaration of Sushi\Sushi::getConnectionName() must be compatible with Spatie\Multitenancy\Models\Tenant::getConnectionName(): string
```

I'm happy to submit a PR to change `UsesLandlordConnection::getConnectionName()` to match Laravel's `Model::getConnectionName()`...but I want to know if I'm barking up the right tree.  🌳  🐕

**Comment by masterix21** [comment]
Hi @telkins. 

IMHO, we should be compatible with the framework, but these changes have been applied by @freekmurze here https://github.com/spatie/laravel-multitenancy/commit/bd4e7ebf42f61cfaa0eb9197ffea9a6a3c6b3784. I prefer to wait for his opinion on it, then if Freek agrees with us, I'll fix it.

Thanks for your time on it.

**Comment by freekmurze** [comment]
I agree that we should be compatible. Feel free to remove that typehint @masterix21 👍

**Comment by masterix21** [comment]
Thanks @freekmurze

**Comment by telkins** [comment]
@masterix21 @freekmurze I just created a quick PR to remove the typehint: #336.

Hope that's OK.  🤓

**Comment by masterix21** [comment]
Merged :) thanks for your time on it.
